packages/frontend/src/hooks/useDynamicData.ts

```diff
@@ -1,40 +1,52 @@
-import type { IField, IFieldDropdownSource, IJSONObject } from '@plumber/types'
+import type { IField, IFieldDropdownSource } from '@plumber/types'
 
-import * as React from 'react'
-import type { UseFormReturn } from 'react-hook-form'
-import { useFormContext } from 'react-hook-form'
-import { useLazyQuery } from '@apollo/client'
+import { useEffect, useMemo } from 'react'
+import { type FieldValues, useFormContext } from 'react-hook-form'
+import { useQuery } from '@apollo/client'
 import { GET_DYNAMIC_DATA } from 'graphql/queries/get-dynamic-data'
-import isEqual from 'lodash/isEqual'
-import set from 'lodash/set'
+import { get, set } from 'lodash'
 
-const variableRegExp = /({.*?})/g
+/**
+ * Map of field path -> argument name
```

Comment:
would the field path and argument name always be the same? cus over here it seems like 'parameters.fileId' => 'parameters.fileId' and may need some clarification on why is this being done?

Reply:
discussed with @ogp-weeloong , this is due for refactor